src/mono/wasm/test-main.js

```diff
@@ -243,53 +242,95 @@ const App = {
 };
 globalThis.App = App; // Necessary as System.Runtime.InteropServices.JavaScript.Tests.MarshalTests (among others) call the App.call_test_method directly
 
+function configureRuntime(dotnet, runArgs, INTERNAL) {
+    dotnet
+        .withVirtualWorkingDirectory(runArgs.workingDirectory)
+        .withEnvironmentVariables(runArgs.environmentVariables)
+        .withDiagnosticTracing(runArgs.diagnosticTracing)
+        .withDiagnosticTracing(true)
+        .withExitOnUnhandledError()
+        .withExitCodeLogging()
+        .withElementOnExit();
+
+    if (is_node) {
+        dotnet
+            .withEnvironmentVariable("NodeJSPlatform", process.platform)
+            .withAsyncFlushOnExit();
+
+        const modulesToLoad = runArgs.environmentVariables["NPM_MODULES"];
+        if (modulesToLoad) {
+            dotnet.withModuleConfig({
+                onConfigLoaded: (config) => {
+                    loadNodeModules(config, INTERNAL.require, modulesToLoad)
+                }
+            })
+        }
+    }
+    if (is_browser) {
+        dotnet.withEnvironmentVariable("IsWebSocketSupported", "true");
+    }
+    if (runArgs.runtimeArgs.length > 0) {
+        dotnet.withRuntimeOptions(runArgs.runtimeArgs);
+    }
+    if (runArgs.debugging) {
+        dotnet.withDebugging(-1);
```

Comment:
How should `withDebugging` and `withWaitingForDebugger` behave for the `dry_run` step? Wouldn't the dry run always hang if it's waiting for debugger, or is it set up so that the create finishes and loading the snapshot will resume waiting for it?

Reply:
I guess this is unit test only internal thing, but maybe part of the debugger tests are failing because of that.
I will disable snapshot if I find `waitForDebugger` set. 

The `withDebugging` is just state of mono initialization with no outside dependency, so it should be OK to snapshot.
I will do more testing.